### ISSUE TITLE
ARM instructions decoded incorrectly after analysis

### ISSUE THREAD

**Issue by MarioVilas** [radare/radare2#8051]
Tested on Raspbian (Raspberry Pi 3, ELF32 file). The following code disassembles correctly if I don't run the analysis (checked against ODA using the same base address just to be sure):

```
[0x00010858]> pd
        |   ;-- entry0:
        |   ;-- section_end..plt:
        |   ;-- section..text:
        |   ;-- $a:
        |   ;-- _start:
        |   0x00010858      00b0a0e3       mov fp, 0                   ; section 13 va=0x00010858 pa=0x00000858 sz=2000 vsz=2000 rwx=--r-x .text
        |   0x0001085c      00e0a0e3       mov lr, 0
        |   0x00010860      04109de4       pop {r1}
        |   0x00010864      0d20a0e1       mov r2, sp
        |   0x00010868      04202de5       str r2, [sp, -4]!
        |   0x0001086c      04002de5       str r0, [sp, -4]!
        |   0x00010870      10c09fe5       ldr ip, loc._d_2            ; [0x10888:4]=0x11024 sym.__libc_csu_fini ; "$\x10\x01"
        |   0x00010874      04c02de5       str ip, [sp, -4]!
        |   0x00010878      0c009fe5       ldr r0, sym.main            ; [0x1088c:4]=0x109e4 sym.main
        |   0x0001087c      0c309fe5       ldr r3, sym.__libc_csu_init ; [0x10890:4]=0x10fc0 sym.__libc_csu_init
        |   0x00010880      b8ffffeb       bl sym.imp.__libc_start_main
        |   0x00010884      e7ffffeb       bl sym.imp.abort
```

However after doing "aaaa" I get this output instead:

```
[0x00010858]> pdf
  |||||||   ;-- section_end..plt:
  |||||||   ;-- section..text:
  |||||||   ;-- $a:
/ (fcn) entry0 (arm) 44
|   entry0 ();
| !!!!!!!      ; UNKNOWN XREF from 0x00010018 (sym.0x00010003 + 21)
| !!!!!!!      ; UNKNOWN XREF from 0x00011614 (section_end..eh_frame + 968)
| !!!!!!!      ; UNKNOWN XREF from 0x00011714 (section_end..eh_frame + 1224)
| !!!!!!!      ; UNKNOWN XREF from 0x00011d04 (section_end..eh_frame + 2744)
| |||||||   0x00010858      00b0           add sp, 0                   ; section 13 va=0x00010858 pa=0x00000858 sz=2000 vsz=2000 rwx=--r-x .text
| ========< 0x0001085a      a0e3           b 0x10f9e                   ; sym.s_puts+0x82
| ========< 0x0001085c      00e0           b 0x10860
| ========< 0x0001085e      a0e3           b 0x10fa2                   ; sym.s_puts+0x86
| --------> 0x00010860      0410           asrs r4, r0, 0x20
| ||||||`=< 0x00010862      9de4           b 0x101a0
| ||||||    0x00010864      0d20           movs r0, 0xd
| ||||||,=< 0x00010866      a0e1           b 0x10baa                   ; sym.main+0x1c6
| |||||||   0x00010868      0420           movs r0, 4
| |||||`==< 0x0001086a      2de5           b 0x102c8
| ||||| |   0x0001086c      0400           movs r4, r0
| ||||`===< 0x0001086e      2de5           b 0x102cc
| ||||  |   0x00010870      10c0           stm r0!, {r4}
| ||`=====< 0x00010872      9fe5           b 0x103b4
| || |  |   0x00010874      04c0           stm r0!, {r2}
| || `====< 0x00010876      2de5           b 0x102d4
| ||    |   0x00010878      0c00           movs r4, r1
| |`======< 0x0001087a      9fe5           b 0x103bc
| |     |   0x0001087c      0c30           adds r0, 0xc
| `=======< 0x0001087e      9fe5           b 0x103c0
|       |   0x00010880      b8ffffeb       invalid
```

Not sure why the disassembly is being forced into Thumb mode, but since I've issued no other commands and the binary is definitely not 16 bits and with the correct virtual address, I'm guessing this must be a bug.

Radare version:
```$ r2 -v
radare2 1.7.0-git 15436 @ linux-arm-32 git.1.6.0-166-gfa576bb
commit: fa576bbb9ab22f7f2842eacbac523aa251cf3a62 build: 2017-07-28__11:15:52
```

**Comment by radare** [comment]
can u share the binary? would be good to know why the analysis changes the bitness of this function you can use afB 32 or afhb 32 to change the bitness f this function

**Comment by radare** [comment]
also, aaaa is usually stupid to run, because its blindly throwing different analysis loops and some may be wrong. can u try which command exactly is making this happen?

**Comment by MarioVilas** [comment]
Here it is:
[stack1.tar.gz](https://github.com/radare/radare2/files/1183033/stack1.tar.gz)

Yeah, I know aaaa is not recommended, I just stumbled upon this while doing some quick tests and I figured it was worth reporting regardless. The afB 32 command does fix the disassembly of course.

**Comment by radare** [comment]
did a quick test. the issue is with aae

**Comment by radare** [comment]
I found the issue and i know how to fix it but had to go for lunch. Will finish the fix after that

**Comment by radare** [comment]
Fixed in aac34821df348e894522ada64047dce4ca767945
